### ISSUE TITLE
Not displaying the full path on the search leads to users clicking on the wrong items

### ISSUE THREAD

**Issue by paoliniluis** [metabase/metabase#49630]
### Describe the bug

Let's say you have the same name of collections in a deeply nested collection structure... when searching for an item in the search bar you might end up selecting the wrong item
![Image](https://github.com/user-attachments/assets/ba1ff24d-914b-4198-af0b-1fe212498776)

### To Reproduce

1) create a "dataset" collection inside our analytics
2) create a dataset collection inside your personal collection
3) create a people question and then save it in the first dataset collection with the "people question" name
4) duplicate the question, save it with the same "people question" name in the second dataset collection
5) try searching for "people question" in the search bar, which one would you pick?

### Expected behavior

we should show the full paths on the search bar when hovering over the collection name

### Logs

NA

### Information about your Metabase installation

-v50+

### Severity

P2

### Additional context

It's annoying

**Comment by luizarakaki** [comment]
I agree with the request, but this is behaving as intended. I'm changing to feature request and assigning to this project [Help users understand and trust search results](https://www.notion.so/metabase/Help-users-understand-and-trust-search-results-15969354c901806f89e8c8fe77016410)
